src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c

```diff
@@ -409,21 +409,33 @@ static void init_resp(struct xgq_com_queue_entry *resp, u16 cid, u32 rcode)
 	memset(resp, 0, sizeof(*resp));
 	resp->hdr.cid = cid;
 	resp->hdr.cstate = XGQ_CMD_STATE_COMPLETED;
+	resp->result = KDS_COMPLETED;
+	resp->rcode = rcode;
+}
+
+static void init_resp_abort(struct xgq_com_queue_entry *resp, u16 cid, u32 rcode, int status)
+{
+	memset(resp, 0, sizeof(*resp));
+	resp->hdr.cid = cid;
+	resp->hdr.cstate = XGQ_CMD_STATE_ABORTED;
+	resp->result = status;
 	resp->rcode = rcode;
 }
 
 static inline void
-zcu_xgq_cmd_complete(struct platform_device *pdev, struct xgq_cmd_sq_hdr *cmd, int ret)
+zcu_xgq_cmd_complete(struct platform_device *pdev, struct xgq_cmd_sq_hdr *cmd, int ret, int status)
 {
 	struct xgq_com_queue_entry r;
+	struct xgq_com_queue_entry *rptr = &r;
 	struct zocl_cu_xgq *zcu_xgq = platform_get_drvdata(pdev);
 
-	if (likely(ret == 0 && ZCU_XGQ_FAST_PATH(zcu_xgq))) {
-		zxgq_send_response(zcu_xgq->zxc_zxgq_hdl, NULL);
-	} else {
+	if (likely(ret == 0 && ZCU_XGQ_FAST_PATH(zcu_xgq)))
+		rptr = NULL;
+	else if (unlikely(status == KDS_SKCRASHED))
+		init_resp_abort(&r, cmd->cid, ret, status);
+	else
 		init_resp(&r, cmd->cid, ret);
```

Comment:
The init_resp and init_resp_aboart are pretty much the same function with the cstate and status be different.

This code can be refactor to:
1. Initialize the r structure with known default values after line 434.
2. Set the values correctly for the KDS_SKCRASHED

Something like:
```
else {
   // Assume no crash
   r.hdr.cid = cid;
   r.result = status;
   r.rcode = rcode;
   r.hdr.cstate = XGQ_CMD_STATE_COMPLETED;  
   r.result = KDS_COMPLETED;
   if (unlikely(status == KDS_SKCRASHED)) {
         r.hdr.cstate = XGQ_CMD_STATE_ABORTED;  
         r.result = status;
   }
}
```